### ISSUE TITLE
Training Speed

### ISSUE THREAD

**Issue by Saltb0xApps** [declare-lab/TangoFlux#14]
Hi there!

I'm pre-training on 4xA100 80GB with the following config -

Num examples = **325480**

Num Epochs = **80**

Instantaneous batch size per device = **9**
Total train batch size (w. parallel, distributed & accumulation) = **108**
Gradient Accumulation steps = **3**

**Total optimization steps = 241120**

The expected time taken for pre-training seems to be around **32 days** with this config.

Is this comparable to TangoFlux's original pre-training setup?

Would it be possible to share wandb/tensorboard or any similar training logs of the original run for guidance?

**Comment by hungchiayu1** [comment]
Yup, our original training on WavCaps took about 1 month too.  If you are training with this dataset size, I think it will be much faster if you precompute the latent and save them. This will require slight modification of code though. In this way, you dont have to load the vae during training, and you can train without gradient accumulation as well.

**Comment by Saltb0xApps** [comment]
@hungchiayu1 would using torch compile + fp16 be a good strategy for further speeding up training? any reason why the original model was not trained with these optimisations too?

**Comment by hungchiayu1** [comment]
Yup, I think it will further speed up the run. Originally I did try to use torch.compile for training but i ran into some strange bug so I end up disabling it. I am currently training with bf16 and it seems to work well.
